src/main/java/org/kiegroup/kogibot/util/WorkflowsUtils.java

```diff
@@ -0,0 +1,97 @@
+                                        .stream()
+                                        .filter(workflowRun -> workflowRun.getHeadRepository().getOwnerName()
+                                                        .equals(pullRequest.getHead().getRepository().getOwnerName()))
+                                        .filter(workflowRun -> workflowRun.getStatus() != GHWorkflowRun.Status.QUEUED
+                                                        || workflowRun.getStatus() != GHWorkflowRun.Status.IN_PROGRESS)
+                                        .collect(Collectors.toList());
+
+                        for (GHWorkflowRun workflowRun : ghWorkflowRuns) {
+                                workflowRun.rerun();
+                        }
+                } else {
+                        log.warn("Issue is not a PR");
+                }
+        }
+
+        /**
+         * Get all workflow runs
+         * 
+         * @param pullRequest
+         * @return
+         * @throws IOException
+         */
+        public static List<GHWorkflowRun> getAllWorkflowRuns(GHPullRequest pullRequest) throws IOException {
+                return pullRequest.getRepository().queryWorkflowRuns()
+                                .branch(pullRequest.getHead().getRef())
+                                .list()
+                                .toSet()
+                                .stream()
+                                .filter(workflowRun -> workflowRun.getHeadRepository().getOwnerName()
+                                                .equals(pullRequest.getHead().getRepository().getOwnerName()))
+                                .collect(Collectors.toList());
+        }
+
+        /**
+         * Return all workflow runs which
+         * 
+         * @param pullRequest
+         * @return
+         * @throws IOException
+         */
+        public static List<GHWorkflowRun> getLatestWorkflowRuns(GHPullRequest pullRequest) throws IOException {
+                Map<String, GHWorkflowRun> latestWorkflowRuns = new HashMap<>();
+                for (GHWorkflowRun wr : getAllWorkflowRuns(pullRequest)) {
+                        if (latestWorkflowRuns.containsKey(wr.getName())) {
+                                if (wr.getRunAttempt() > latestWorkflowRuns.get(wr.getName()).getRunAttempt()) {
+                                        latestWorkflowRuns.put(wr.getName(), wr);
+                                }
+                        } else {
+                                latestWorkflowRuns.put(wr.getName(), wr);
+                        }
+                }
+                return latestWorkflowRuns.values().stream().collect(Collectors.toList());
+        }
+
+        /**
+         * Return all workflow runs which
+         * 
+         * @param pullRequest
+         * @return
+         * @throws IOException
+         */
+        public static List<GHWorkflowRun> getNonRunningWorkflows(GHPullRequest pullRequest) throws IOException {
```

Comment:
See the description at the top of the class, the `/test` or `/retest` or `/bot test` would have 3 possibilities:
- Nothing after the command -> Restart all
- `failed` is added, like `/retest failed` -> This would restart only failed workflows
- workflow name is added -> This would restart only the given workflow

PR is still draft as I did not finish some of the commands and some tests are missing. But I will add some more next week probably